src/app/shared/components/managecitations/managecitations.component.ts

```diff
@@ -0,0 +1,112 @@
+// ------------------------------------------------------------------------------
+// ----- atLeast1Req.modal.ts ---------------------------------------------------
+// ------------------------------------------------------------------------------
+
+// copyright:   2017 WiM - USGS
+// authors:  Tonia Roddick USGS Wisconsin Internet Mapping
+// purpose: modal used to show about information
+
+import { Component, ViewChild, OnInit, OnDestroy } from '@angular/core';
+import { NgbModal} from '@ng-bootstrap/ng-bootstrap';
+import { NSSService } from 'app/shared/services/app.service';
+import { SettingsService } from 'app/settings/settings.service';
+import { Config } from 'app/shared/interfaces/config';
+import { ConfigService } from 'app/config.service';
+import { Scenario } from 'app/shared/interfaces/scenario';
+import { ToasterService } from 'angular2-toaster';
+import { AuthService } from 'app/shared/services/auth.service';
+import { Citation } from 'app/shared/interfaces/citation';
+
+@Component({
+    selector: 'manageCitationsModal',
+    templateUrl: './managecitations.component.html',
+    styleUrls: ['./managecitations.component.css']
+})
+export class ManageCitationsModal implements OnInit, OnDestroy {
+    @ViewChild('manageCitations') public manageCitationsModal; // : ModalDirective;  //modal for validator
+    private modalElement: any;
+    private modalSubscript;
+    public appVersion: string;
+    public regressionRegions;
+    public selectedRegion;
+    private configSettings: Config;
+    public modalRef;
+    public loggedInRole;
+    public citations: Array<Citation>;
+    public scenarios: Scenario[];
+
+    constructor(private _nssService: NSSService, private _modalService: NgbModal,
+        private _settingsService: SettingsService, private _configService: ConfigService, private _toasterService: ToasterService,
+        private _authService: AuthService) {
+        this.configSettings = this._configService.getConfiguration();
+    }
+
+    ngOnInit() {
+        // subscriber for logged in role
+        this.loggedInRole = localStorage.getItem('loggedInRole');
+        this._authService.loggedInRole().subscribe(role => {
+            if (role === 'Administrator' || role === 'Manager') {
+                this.loggedInRole = role;
+            }
+        });
+        this.modalSubscript = this._nssService.showManageCitationsModal.subscribe((show: boolean) => {
+            if (show) { this.showModal(); }
+        });
+        this._nssService.selectedRegion.subscribe(region => {
+            this.selectedRegion = region;
+            if (region && region.id) {this.getRegRegions(); }
+        });
+        this._nssService.getVersion.subscribe((v: string) => {
```

Comment:
I think this can be taken out as well.

Reply:
Which lines?

Reply:
Oh yeah, that looks confusing. Starting at 59, that function where it subscribes to the appVersion. I don't think the appVersion is used. You can take that out of the variables as well.